pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -2467,6 +2492,13 @@ func (gce *GCECloud) getInstancesByNames(names []string) ([]*gceInstance, error)
 			pageToken = res.NextPageToken
 			for _, i := range res.Items {
 				name := i.Name
+				if inst, ok := instances[name]; !ok || inst != nil {
+					if inst != nil {
+						glog.Warningf("GCE returned duplicate result for instance %q", name)
```

Comment:
I added it when was debugging that internal bug. I was thinking there was instability across pages (I.e. with the wrong options it returned `a,b,c` then `c,d,f`) in some race condition, which would then show up as a missing instance a few lines below (so the idea is that the pair of lines would show the instability). That ended up not being the case, but it doesn't seem harmful to leave in.